src/manage.c

```diff
@@ -171,8 +183,16 @@ static void manage_calc_almanac_scores(void)
       /* No almanac or position/time information, give it the benefit of the
        * doubt. */
       acq_prn_param[prn].score[ACQ_HINT_ALMANAC] = 0;
+    } else if (almanac[prn].valid && time_quality > TIME_COARSE &&
+              (abs(almanac[prn].week - t.wn%1024) > MAX_ALM_WEEK_DIFF)) {
```

Comment:
Also we ought to make sure the week number stuff is thoroughly thought through.  IIRC the policy was to use full-length, non-wrapping week numbers internally, and correct the received mod-1024 value to the full length value at the time we receive a TOW, ephemeris or almanac from the sky or the internet.  So the % 1024 here, and the -1024 later, might not be appropriate here.  @fnoble is that what you had in mind?